cmd/createorupdate/createorupdate.go

```diff
@@ -272,50 +343,35 @@ func main() {
 		rpURL = sdk.DefaultBaseURI
 	}
 	rpc := sdk.NewOpenShiftManagedClustersClientWithBaseURI(rpURL, os.Getenv("AZURE_SUBSCRIPTION_ID"))
+	authorizer, err := azureclient.NewAuthorizer(os.Getenv("AZURE_CLIENT_ID"), os.Getenv("AZURE_CLIENT_SECRET"), os.Getenv("AZURE_TENANT_ID"))
+	if err != nil {
+		log.Fatal(err)
+	}
+	rpc.Authorizer = authorizer
 
 	if strings.ToUpper(*method) == http.MethodDelete {
-		future, err := rpc.Delete(context.Background(), os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"))
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		if err := future.WaitForCompletionRef(context.Background(), rpc.Client); err != nil {
-			logrus.Fatal(err)
-		}
-		resp, err := future.Result(rpc)
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		if resp.StatusCode != http.StatusOK {
-			logrus.Fatalf("unexpected status: %s, expected 200 OK", resp.Status)
+		if err := delete(ctx, log, rpc); err != nil {
+			log.Fatal(err)
 		}
 		return
 	}
 
 	// simulate the API call to the RP
-	in, err := ioutil.ReadFile("_data/manifest.yaml")
-	if err != nil {
-		logrus.Fatal(err)
-	}
-	var oc sdk.OpenShiftManagedCluster
-	if err := yaml.Unmarshal(in, &oc); err != nil {
-		logrus.Fatal(err)
-	}
-	future, err := rpc.CreateOrUpdate(context.Background(), os.Getenv("RESOURCEGROUP"), os.Getenv("RESOURCEGROUP"), oc)
-	if err != nil {
-		logrus.Fatal(err)
-	}
-	if err := future.WaitForCompletionRef(context.Background(), rpc.Client); err != nil {
-		logrus.Fatal(err)
-	}
-	resp, err := future.Result(rpc)
-	if err != nil {
-		logrus.Fatal(err)
+	if err := createOrUpdate(ctx, log, rpc, *manifest); err != nil {
+		log.Fatal(err)
 	}
-	out, err := yaml.Marshal(resp)
-	if err != nil {
-		logrus.Fatal(err)
+
+	// if an update is requested, do it
+	if *update != "" {
+		if err := createOrUpdate(ctx, log, rpc, *update); err != nil {
+			log.Fatal(err)
+		}
 	}
-	if err := ioutil.WriteFile("_data/manifest.yaml", out, 0666); err != nil {
-		logrus.Fatal(err)
+
+	// if a cleanup is requested, do that too
+	if *cleanup {
```

Comment:
Should be deferred but can be done in a follow-up.

Reply:
https://github.com/openshift/openshift-azure/issues/644

Reply:
#646